spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingPlugin.java

```diff
@@ -100,20 +102,22 @@ public void postProcessModule(Module module) {
 			}
 			LocalMessageBusHolder messageBusHolder = new LocalMessageBusHolder();
 			messageBusHolder.set(module.getComponent(MessageBus.class));
-			receiver = new MessageBusReceiver(messageBusHolder, storageLevel, messageBusProperties, inboundModuleProperties);
+			receiver = new MessageBusReceiver(messageBusHolder, storageLevel, messageBusProperties,
+					inboundModuleProperties, getMimeType(module, true));
```

Comment:
Initially I went through this route. But, I would have to set either of these to empty string when calling the method as we only need either the input or output MIME type. or, do you mean some other way?